src/providers/Context/Context.provider.jsx

```diff
@@ -13,29 +15,59 @@ function useVideosUpdate() {
   return useContext(VideoUpdateContext);
 }
 
+function useVideoDetailed() {
+  return useContext(VideoDetailedContext);
+}
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'grid':
+      return { videos: action.payload.data, video: null };
+    case 'detail':
+      return { videos: null, video: action.payload.data };
+    default:
+      return state;
+  }
+}
+
 const ContextProvider = ({ children }) => {
-  const [videos, setVideos] = useState(dataInit);
+  const [state, dispatch] = useReducer(reducer, { videos: dataInit });
 
   function searchVideos(e) {
     if (e.key === 'Enter') {
       e.preventDefault();
 
       if (e.target.value.length === 0) {
-        setVideos(dataInit);
+        dispatch({ type: 'grid', payload: { data: dataInit } });
       } else {
-        setVideos(dataSearch);
+        const response = youtube
+          .search(e.target.value)
+          .then((response) => {
+            dispatch({ type: 'grid', payload: { data: response.data } });
+          })
+          .catch((err) => {
+            console.log(err);
+          });
       }
     }
   }
 
+  function selectVideo(e, data) {
+    e.preventDefault();
+    const video = data;
+    dispatch({ type: 'detail', payload: { data: video } });
+  }
+
   return (
-    <VideoContext.Provider value={videos}>
+    <VideoContext.Provider value={state}>
       <VideoUpdateContext.Provider value={searchVideos}>
-        {children}
+        <VideoDetailedContext.Provider value={selectVideo}>
+          {children}
+        </VideoDetailedContext.Provider>
       </VideoUpdateContext.Provider>
     </VideoContext.Provider>
```

Comment:
It seems that is correctly functioning but a few questions for you to think:
1. Is there a specific reason for using 3 providers?
2. Is it doable with a single provider?
3. If so, can it be more readable for you or others?